### PR TITLE
feat: allow movable `workerData`

### LINKED ISSUE #26: Support movables in `workerData` option
Two feature requests here:

1. Passing `workerData: Tinypool.move(data)` is not supported, even though `new Worker` has a `transferList` option.

2. If `workerData` is a plain object, Tinypool should iterate its property values to see if any of them are movable, thereby allowing multiple movables to be passed through the `workerData` option.

### PR DESCRIPTION
**Pull request by aleclarson**
Closes #26

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1036,6 +1009,35 @@ class Tinypool extends EventEmitterAsyncResource {
   }
 }
 
+/**
+ * Handle transferable `data` or transferable properties of `data` by unwrapping
+ * them and adding their transferable objects to the given `transferList` array.
+ */
+function fillTransferList(data: any, transferList: TransferListItem[]): any {
```
Comment (aleclarson) [inline src/index.ts:1016]: Here's the star of the show. The rest is mainly TypeScript improvements.

File: src/index.ts [hunk 2]
```diff
@@ -254,23 +245,8 @@ class TaskInfo extends AsyncResource implements Task {
   ) {
     super('Tinypool.Task', { requireManualDestroy: true, triggerAsyncId })
     this.callback = callback
-    this.task = task
+    this.task = fillTransferList(task, transferList)
```
Comment (aleclarson) [inline src/index.ts:248]: This PR also adds nested movable support to the `pool.run` method. ☝️ 

Note that only 1 level of nesting is checked. In other words, doing this 👇 is not supported:
```ts
pool.run({ a: { b: Tinypool.move(b) } })
```

File: src/index.ts [hunk 3]
```diff
@@ -3,6 +3,7 @@ import {
   MessageChannel,
   MessagePort,
   receiveMessageOnPort,
+  TransferListItem,
```
Comment (aleclarson) [inline src/index.ts:6]: Turns out `worker_threads` exports a `TransferListItem` type for us, so we don't have to extract it from `MessagePort.postMessage` anymore :)

**Comment by Aslemammad** [comment]
@aleclarson Thank you, Alec, This looks good overall, though, we need to add tests for it and a short description in README!
It'd be good if you could add those so I can review them again!

**Comment by Aslemammad** [comment]
@aleclarson Any updates on this?

**Pull request closed without merging**